CMakeLists.txt

```diff
@@ -696,14 +697,23 @@ else()
 endif()
 
 # add enabled sanitizers
-function(add_sanitizer sanitizer supported_compilers want_flag status_flag)
+function(add_sanitizer sanitizer supported_compilers want_flag status_flag build_type)
 	if(${want_flag})
 		if(CMAKE_CXX_COMPILER_ID MATCHES "${supported_compilers}")
+			if (build_type STREQUAL "debug")
+				set(C_FLAGS CMAKE_C_FLAGS_DEBUG)
+				set(CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG)
+			elseif(build_type STREQUAL "any")
+				set(C_FLAGS CMAKE_C_FLAGS)
+				set(CXX_FLAGS CMAKE_CXX_FLAGS)
+			endif()
+
 			set("${status_flag}" "Enabled" PARENT_SCOPE)
 			string(REPLACE ";" " " additional_flags "${ARGN}")
+
 			# todo CMake 3.13: use add_compile_options/add_link_options instead
-			set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=${sanitizer} ${additional_flags}" PARENT_SCOPE)
-			set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=${sanitizer} ${additional_flags}" PARENT_SCOPE)
+			set(${C_FLAGS} "${${C_FLAGS}} -fsanitize=${sanitizer} ${additional_flags}" PARENT_SCOPE)
```

Comment:
We require at least CMake 3.13 now, so maybe this would be a good time to fix the TODO comment on the line above?